docs/inventories.rst

```diff
@@ -0,0 +1,773 @@
+
+Also, when a variable may or may not be defined (depending on the host), you can give a default, like with ``| default("running")`` if you know what value it should have when it is not defined. This helps with debugging, as described in the :ref:`constr_inv_debugging` section.
+
+2. **Construct 1 group, limit to group**
+
+``source_vars``:
+
+::
+
+  plugin: constructed
+  strict: true
+  groups:
+    shutdown_in_product_dev: state | default("running") == "shutdown" and account_alias == "product_dev"
+
+
+``limit``: ``shutdown_in_product_dev``
+
+This input creates one group that only includes hosts that match both criteria. The limit is then just the group name by itself, returning just **host2**, same as the previous approach.
+
+.. _constr_inv_debugging:
+
+Debugging tips
+^^^^^^^^^^^^^^
+
+It is very important to set the ``strict`` parameter to ``True`` so that you can debug problems with your templates. If the template fails to render, you will get an error in the associated inventory update for that constructed inventory.
+
+When encountering errors, increase verbosity to get more details.
+
+Giving a default, like with ``| default("running")`` is a generic use of Jinja2 templates in Ansible. Doing this avoids errors from the particular template when you set ``strict: true``. You could also set ``strict: false``, and allow the template to produce an error, which results in the host not getting included in that group. However, doing so makes it difficult to debug issues in the future if your templates continue to grow in complexity.
+
+However, you may still have to debug the intended function of the templates if they are not producing the expected inventory content. For example, if a ``groups`` group has a complex filter (like ``shutdown_in_product_dev``) but does not contain any hosts in the resultant constructed inventory, then use the ``compose`` parameter to help debug. Like this:
+
+``source_vars``:
+
+::
+
+  plugin: constructed
+  strict: true
+  groups:
+    shutdown_in_product_dev: state | default("running") == "shutdown" and account_alias == "product_dev"
+  compose:
+    resolved_state: state | default("running")
+    is_in_product_dev: account_alias == "product_dev"
+
+``limit``: ``
+
+.. |bt| raw:: html
+
+    <code class="code docutils literal notranslate">``</code>
+
+Running with a blank ``limit`` will return all hosts. You can use this to inspect specific variables on specific hosts, giving insight into where problems in the ``groups`` lie.
+
+
+Nested groups
+~~~~~~~~~~~~~
+
+The inventory contents of two groups where one is a child of the other is used here to demonstrate nested groups. The child group has a host inside of it, and the parent group has a variable defined. Due to how Ansible core works, the variable of the parent group will be available in the namespace as a playbook is running, and can be used for filtering.
+
+Define the inventory file in a yaml format named ``nested.yml``:
+
+::
```

Comment:
lexer missing

Reply:
Should it be this?
```
:file:`nested.yml`
```

Reply:
No, that's a reference.
A lexer is a thing used for syntax highlighting. Essentially, you should tell RST/Sphinx what markup this is so it'd be able to color the code block correctly.

Reply:
Thanks for the suggestions. Since tests are passing, and this work can be separated out as an independent task, let's revisit this when I am more familiar with what these elements are you want applied and put all the code block changes in separate PR at a later time.